### ISSUE TITLE
[1.1.0] iOS: Exception when backgrounding the app

### ISSUE THREAD

**Issue by simonsturge** [achorein/expo-share-intent#26]
In the new 1.1.0 version, the app is crashing when backgrounding with the following log:

```
 DEBUG  useShareIntent[to-background] reset intent
 ERROR  Error: FunctionCallException: Calling the 'clearShareIntent' function has failed (at ExpoModulesCore/SyncFunctionComponent.swift:87)
→ Caused by: ArgumentCastException: The 1st argument cannot be cast to type String (at ExpoModulesCore/JavaScriptUtils.swift:40)
→ Caused by: NullCastException<String>: Cannot cast null to non-optional 'String' (at ExpoModulesCore/DynamicRawType.swift:25), js engine: hermes
 DEBUG  useShareIntent[active] refresh intent
 DEBUG  useShareIntent[refresh] null
```

![IMG_C40EFDDA22AC-1](https://github.com/achorein/expo-share-intent/assets/3878145/7bdfc6e1-9ac3-435a-9ea7-0f62a50698a9)

**Comment by achorein** [comment]
do you use a "static" build ?

**Comment by achorein** [comment]
I cannot reproduce your issue :/

can you try this :
- disable auto clean on background :
```ts
const { shareIntent, resetShareIntent } = useShareIntent({ resetOnBackground: false });
```
- handle manualy the reset : 
```ts
import { clearShareIntent } from "expo-share-intent";
```
```ts
  useEffect(() => {
    const subscription = AppState.addEventListener("change", (nextAppState) => {
      if (appState.current === "active" &&
        ["inactive", "background"].includes(nextAppState)
      ) {
        console.debug("to-background");
        clearShareIntent(`${Constants.expoConfig?.scheme}ShareKey`);
      }
      appState.current = nextAppState;
    });
    return () => {
      subscription.remove();
    };
  }, []);
```

if it display "to-background" and not crash i can do a patch...

**Comment by simonsturge** [comment]
> do you use a "static" build ?

Yes, it's a requirement of Firebase I believe

> if it display "to-background" and not crash i can do a patch...

it does display "to-background", no crash when handled manually

Thanks!

**Comment by achorein** [comment]
fixed in v1.1.1 and v0.3.1.
fill free to reopen if the problem persist
